src/pages/films.js

```diff
@@ -1,54 +1,113 @@
+import Layout from '../components/layout'
+
+
+export default function Films({ }) {
+
+  const [data, setData] = useState([]);
+  const [filmFilter, setFilmFilter] = useState("all");
+  const [films, setFilms] = useState([]);
+  const [searchYear, setSearchYear] = useState('')
 
-export const query = graphql`
-    query FilmQuery{
-        hasura {
-            tnff_films {
-                id
-                title
-                tnff_year
-                location
-                category
-                release_date
-                duration_mins
-                language
-                director
-                producer
-                synopsis
-                director_bio
-                film_img_url
-                director_img_url
-                other_info
-            }
-        }
+
+  // ! breakdown tasks/lil sprints for timeline
+
+  const getData = useCallback(async function () {
+    try {
+      const data = await fetch("https://sonal-hasura.herokuapp.com/v1/graphql", {
+        "method": "POST",
+        "headers": {
+          "Content-Type": "application/json"
+        },
+        "body": "{\"query\":\"{\\n tnff_films (order_by: {tnff_year: desc}) {\\n                id\\n                title\\n                tnff_year\\n                location\\n                category\\n                release_date\\n                duration_mins\\n                language\\n                director\\n                producer\\n                synopsis\\n                director_bio\\n                film_img_url\\n                director_img_url\\n                other_info\\n            }\\n}\"}"
+      });
+      const json = await data.json();
+      setData(json.data.tnff_films);
+  
+    } catch (err) {
+      setData([]);
+      console.log(err);
     }
-`
+  }, []);
+
+
+  useEffect(() => {
+    getData();
+
+  }, [getData]);
+
+  // Set filter in state for given input
+  const filter = (e) => {
+    console.log(e.target.textContent)
+    setFilmFilter(e.target.textContent)
+  };
+
+
+  // let filmsToRender = [];
+  // data.forEach((film) => {
+  //   if (filmFilter === "all") {
+  //     filmsToRender.push(film);
+  //     // console.log(film)
+  //     //   filmsToRender.push(film);
+  //   } else {
+  //     // console.log(data)
+  //     data.forEach(film => {
+  //       console.log(film)
+        // if (film.tnff_year === filmFilter) {
+      //     console.log(filmFilter)
+      //   console.log(film);
+      //   // filmsToRender.push(film)
+      //   }
+//       });
+//     }
+//   })
+// console.log(filmsToRender)
+// console.log(data)
+
+// gallery with film image and title
+const filmData = data.map(film => {
+const { film_img_url, title, id, release_date } = film;
+if (film_img_url) {
+  return (
+      <div className="film_single" key={id}>
+          <img src={`${film_img_url}`} alt="" />
+          <div className="movie_data">
+              <h2>{title}</h2>
+              <p>{ release_date}</p>
```

Comment:
can you use tnff_year for the filter, the release date of the film doesn't correspond to what year it was shown at the festival